platform/core/src/extensions/ExtensionManager.js

```diff
@@ -40,13 +40,12 @@ export default class ExtensionManager {
       _extensionLifeCycleHooks,
     } = this;
 
-    const {
-      MeasurementService,
-      ViewportGridService,
-    } = _servicesManager.services;
+    const { MeasurementService } = _servicesManager.services;
 
+    for (const service of Object.values(_servicesManager.services)) {
+      service?.onModeEnter?.();
+    }
```

Comment:
To me a Service should not know anything about what is a `mode`. I understand it should clean up itself or init itself. But one can do that on a mode config `onModeEnter` and `onModeExit` like here  https://github.com/OHIF/Viewers/blob/v3-stable/modes/tmtv/src/index.js#L137-L150

Reply:
That breaks the encapsulation between modes and services.  A mode on enter shouldn't know about any services which it doesn't specifically configure/manage.  That is, one should be able to add new services without touching every single mode if that service needs mode setup.  That should ALSO be true for AppInit, that is, there should be an 'onAppInit' call in the service that allows for service specific instantiation.

Putting service specific mode setup/cleanup into the Mode.tsx breaks the encapsulation between the Modes.tsx and the service.  Imagine if a service needed mode setup in the cornerstonejs extension.  Now you need to modify EVERY mode using the cornerstone extension.  That is what I mean by breaking encapsulation.  It would also require tweaking the Modes.tsx file to call the initialization service on setup, so you need an @ohif/ui change in order to call it non-generically from the Modes.tsx.

Reply:
I still having hard time getting convinced this is actually a good idea, for the same reasons we talked e.g., in certain use cases like switching modes and keeping measurements measurementService should not reset. 

So then I don't see the benefit of splitting services to some that has onModeEnter and some that don't. Summoning @swederik